OpenMarket/OpenMarket/ViewController/RegisterView/RegistrationViewController.swift

```diff
@@ -0,0 +1,283 @@
+//
+//  RegistrationViewController.swift
+//  OpenMarket
+//
+//  Created by 써니쿠키, 메네 on 2022/12/01.
+//
+
+import UIKit
+
+final class RegistrationViewController: UIViewController {
+    private let registrationView: RegistrationView = RegistrationView()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.view = registrationView
+        setupNavigationBar()
+        controlKeyBoard()
+        
+        registrationView.textView.delegate = self
+        registrationView.mainScrollView.delegate = self
+        registrationView.imageCollectionView.delegate = self
+        registrationView.imageCollectionView.dataSource = self
+        registrationView.imageCollectionView.register(
+            ImageCollectionViewCell.self,
+            forCellWithReuseIdentifier: ImageCollectionViewCell.identifier
+        )
+    }
+    
+    private func setupNavigationBar() {
+        let button = UIBarButtonItem(title: "Done",
+                                     style: .plain,
+                                     target: self,
+                                     action: #selector(registerProduct))
+        
+        navigationItem.title = "상품등록"
+        navigationItem.rightBarButtonItem  = button
+    }
+    
+    @objc private func registerProduct() {
+        let marketURLSessionProvider = MarketURLSessionProvider()
+        let encoder = JSONEncoder()
+        let product = createProductFromUserInput()
+        let images = registrationView.selectedImages
+        
+        guard let productData = try? encoder.encode(product) else { return }
+        
+        guard let request = marketURLSessionProvider.generateRequest(
+            textParameters: ["params": productData],
+            imageKey: "images",
+            images: images) else { return }
+        
+        marketURLSessionProvider.uploadProduct(request: request) { result in
+            switch result {
+            case .success(_):
+                DispatchQueue.main.async {
+                    self.navigationController?.popViewController(animated: true)
+                }
+            case .failure(_):
+                CustomAlert.showAlert(message: "업로드 실패입니다. 다시 시도해 주세요", target: self)
+            }
+        }
+    }
```

Comment:
```suggestion
        marketURLSessionProvider.uploadProduct(request: request) { result in
            switch result {
            case .success:
                DispatchQueue.main.async {
                    self.navigationController?.popViewController(animated: true)
                }
            case .failure:
                CustomAlert.showAlert(message: "업로드 실패입니다. 다시 시도해 주세요", target: self)
            }
        }
    }
```
열거형의 인자가 사용되지 않는다면 위와 같이 사용하는게 더 깔끔할지도요~~ :)

Reply:
감사합니다🙇🏻‍♀️ 

수정커밋: [`1af9d3f`](https://github.com/yagom-academy/ios-open-market/pull/236/commits/1af9d3fdb467ffdcebf65ec15adbee4a6814502b)